src/compiler/binder.ts

```diff
@@ -1072,7 +1072,6 @@ namespace ts {
                 node = node.parent;
             }
             return !isStatementCondition(node) &&
-                !isLogicalAssignmentExpression(node.parent) &&

```

Comment:
This function is called when `node` was a LogicalAssignmentOperator. But from the tests, I guess that the failing case was when `node` was a LogicalOperator` and `node.parent` was a LogicalAssignmentOperator. I don't understand why the control flow doesn't get created when the parent LogicalAssignmentOperator is `node`.

Also, are there now more control flow nodes created for other nodes besides LogicalAssignmentOperators? I'll run the perf tests.

Reply:
Much later: I was wrong. It is called when `node` is a QuestionQuestionToken and `node.parent` is a LogicalAssignmentExpression. From the new test, when node is `x` in `d ??= x ?? "string"`, and `node.parent` is `??=`.

I guess the deleted line was added because `??=` is a logical expression...but it's also an assignment expression.